test/org/zaproxy/zap/extension/pscanrules/CacheControlScannerUnitTest.java

```diff
@@ -0,0 +1,348 @@
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void httpsBadPragmaCacheRequest() throws HttpMalformedHeaderException {
+        
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Cache-Control: no-cache, no-store, must-revalidate, private\r\n" +
+                "Pragma: cache\r\n" +
+                "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getParam(), equalTo(HttpHeader.PRAGMA));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("cache"));
+    }
+
+    @Test
+    public void httpsRedirectLowCacheRequest() throws HttpMalformedHeaderException {
+        
+        rule.setLevel(AlertThreshold.LOW);
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 301 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Location: http://www.example.org/test2\r\n" +
+                "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getParam(), equalTo(HttpHeader.CACHE_CONTROL));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo(""));
+    }
+    
+    @Test
+    public void httpsRedirectMedCacheRequest() throws HttpMalformedHeaderException {
+        
+        rule.setLevel(AlertThreshold.MEDIUM);
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 301 OK\r\n" +
```

Comment:
Is the status text at all relevant? By spec/convention shouldn't this actually be `HTTP/1.1 301 Moved Permanently`?